pkg/build/controller/build/build_controller.go

```diff
@@ -350,6 +351,16 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 		}
 	}
 
+	// set the pushSecret that will be needed by the build to push the image to the registry
+	// at the end of the build.
+	build.Spec.Output.PushSecret = secret
+
+	// ensure the build object the pod sees starts with a clean set of reasons/messages,
+	// rather than inheriting the potential "invalidoutputreference" message from the current
+	// build state.
+	build.Status.Reason = ""
```

Comment:
Does this code actually have any effect?  If not, I appreciate the belt and braces, but I'm worried that it will confuse more than it will protect.  I think it should be removed if possible.